File: Sample/SampleOC/ViewController.m
//
//  ViewController.m
//  SampleOC
//
//  Created by KK on 2026/7/4.
//

#import "ViewController.h"

#import "SampleAlertViewController.h"

@import OverlayTransitionKit;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)presentButtonAction:(UIButton *)sender {
    __weak typeof(self) weakSelf = self;
    SampleAlertViewController *alertViewController = [SampleAlertViewController new];
    alertViewController.didTapCloseHandler = ^(SampleAlertViewController *_Nonnull alertViewController) {
        // 1.
        //        [weakSelf dismissViewControllerAnimated:YES completion:^{
        //
        //        }];

        // 2.
        [alertViewController dismissViewControllerAnimated:YES completion:^{

        }];
    };
    NSLog(@"otk_presentOverlay begin");
    [self otk_presentOverlay:alertViewController completion:^{
        NSLog(@"otk_presentOverlay completion");
    }];
}

- (IBAction)showChildButtonAction:(UIButton *)sender {
    __weak typeof(self) weakSelf = self;
    SampleAlertViewController *alertViewController = [SampleAlertViewController new];
    alertViewController.didTapCloseHandler = ^(SampleAlertViewController *_Nonnull alertViewController) {
        NSLog(@"otk_dismissChildOverlay begin");

        // 1.
        //        [weakSelf otk_dismissChildOverlay:^{
        //            NSLog(@"otk_dismissChildOverlay completion");
        //        }];

        // 2.
        //        [weakSelf otk_dismissChildOverlay:alertViewController completion:^{
        //            NSLog(@"otk_dismissChildOverlay completion");
        //        }];

        // 3.
        [alertViewController otk_dismissChildOverlay:^{
            NSLog(@"otk_dismissChildOverlay completion");
        }];
    };
    NSLog(@"otk_showChildOverlay begin");
    [self otk_showChildOverlay:alertViewController completion:^{
        NSLog(@"otk_showChildOverlay completion");
    }];
}
@end

File: Sample/Sample/ViewController.swift
//
//  ViewController.swift
//  Sample
//
//  Created by KK on 2026/7/4.
//

import OverlayTransitionKit
import UIKit

class ViewController: UIViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func presentButtonAction(_ sender: UIButton) {
        let alertViewController = SampleAlertViewController()
        alertViewController.didTapCloseHandler = { [weak self] alertViewController in
            // 1.
            // self?.dismiss(animated: true) {
            //
            // }

            // 2.
            alertViewController.dismiss(animated: true) {}
        }
        print("presentOverlay begin")
        presentOverlay(alertViewController) {
            print("presentOverlay completion")
        }
    }

    @IBAction func showChildButtonAction(_ sender: UIButton) {
        let alertViewController = SampleAlertViewController()
        alertViewController.didTapCloseHandler = { [weak self] alertViewController in
            print("dismissChildOverlay begin")

            // 1.
            // self?.dismissChildOverlay {
            //     print("dismissChildOverlay completion")
            // }

            // 2.
            // self?.dismissChildOverlay(alertViewController) {
            //     print("dismissChildOverlay completion")
            // }

            // 3.
            alertViewController.dismissChildOverlay {
                print("dismissChildOverlay completion")
            }
        }
        print("showChildOverlay begin")
        showChildOverlay(alertViewController) {
            print("showChildOverlay completion")
        }
    }
}
